internal/resources/providers/aws_cis/monitoring/filter_pattern_parser.go

```diff
@@ -0,0 +1,275 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package monitoring
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+const maxDepth = 5
+
+type logicalOperator string
+type comparisonOperator string
+
+const (
+	loAnd logicalOperator = "&&"
+	loOr  logicalOperator = "||"
+
+	coEqual     comparisonOperator = "="
+	coNotEqual  comparisonOperator = "!="
+	coNotExists comparisonOperator = "NOT EXISTS"
+)
+
+func listLogicalOperators() []logicalOperator {
+	return []logicalOperator{loAnd, loOr}
+}
+
+func listComparisonOperator() []comparisonOperator {
+	// This order must be kept because we need to check first different and then equals
+	return []comparisonOperator{coNotExists, coNotEqual, coEqual}
+}
+
+// MetricFilterPattern is a union between simpleExpression and complexExpression
+//
+//	because it can be either one or the other
+type MetricFilterPattern struct {
+	simpleExpression  //nolint:unused
+	complexExpression //nolint:unused
```

Comment:
can you explain why did you disable linting here?

Reply:
I'll update the comment to better reflect.

But this `MetricFilterPattern` struct is never actually accessed inside go layer, only in opa layer.

Ideally we would have an interface that both `simpleExpression` and `complexExpression` implements, but because there is actually no need for a method, just a "type holder" I can't do it in go. The simplest way I found to do it was by using embedded fields, acting as a union of both types.